### PR TITLE
Add SAML options to auth0_connection resource

### PR DESCRIPTION
**Pull request by kgunbin**
Based on https://github.com/go-auth0/auth0/pull/118

### Proposed Changes

* Add SAML options to auth0_connection resource. Only support the options that are both readable and writable.
* Added an acceptance test and documentation

#### Acceptance Test Output

```
$ make testacc TESTS=TestAccConnectionSAML
==> Checking that code complies with gofmt requirements...
?   	github.com/terraform-providers/terraform-provider-auth0	[no test files]
=== RUN   TestAccConnectionSAML
--- PASS: TestAccConnectionSAML (1.82s)
PASS
coverage: 10.8% of statements
ok  	github.com/terraform-providers/terraform-provider-auth0/auth0	1.845s	coverage: 10.8% of statements
?   	github.com/terraform-providers/terraform-provider-auth0/auth0/internal/debug	[no test files]
testing: warning: no tests to run
PASS
coverage: 0.0% of statements
ok  	github.com/terraform-providers/terraform-provider-auth0/auth0/internal/random	0.023s	coverage: 0.0% of statements [no tests to run]
testing: warning: no tests to run
PASS
coverage: 0.0% of statements
ok  	github.com/terraform-providers/terraform-provider-auth0/auth0/internal/validation	0.009s	coverage: 0.0% of statements [no tests to run]
?   	github.com/terraform-providers/terraform-provider-auth0/version	[no test files]
```

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

File: auth0/resource_auth0_connection.go [hunk 1]
```diff
@@ -429,6 +429,51 @@ var connectionSchema = map[string]*schema.Schema{
...
+				"binding_method": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "The SAML Response Binding: how the SAML token is received by Auth0 from IdP",
+					ValidateFunc: validation.StringInSlice([]string{
+						"urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect",
+						"urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST",
+					}, true),
+				},
+				"sign_in_endpoint": {
```
Comment (jbauth0) [inline auth0/resource_auth0_connection.go:446]:
```suggestion
				"idpinititated":{
				     Type:     schema.TypeList,
				     MaxItems: 1,
				     Required: false,
				    Elem: &schema.Resource{
					Schema: map[string]*schema.Schema{
						"client_id": {
							Type:     schema.TypeString,
							Optional: true,
						},
						"client_protocol": {
							Type:      schema.TypeString,
							Optional:  true,
                                                },
                                                "client_authorizequery":{
                                                        Type:      schema.TypeString,
							Optional:  true,
					      }
                                                
					}
				},
				"sign_in_endpoint": {
```

File: auth0/structure_auth0_connection.go [hunk 2]
```diff
@@ -200,6 +202,23 @@ func flattenConnectionOptionsAzureAD(o *management.ConnectionOptionsAzureAD) int
 	}
 }
 
+func flattenConnectionOptionsSAML(o *management.ConnectionOptionsSAML) interface{} {
+	return map[string]interface{}{
+		"signing_cert":        o.GetSigningCert(),
+		"protocol_binding":    o.GetProtocolBinding(),
+		"debug":               o.GetDebug(),
+		"idpinitiated":        o.GetIdpInitiated(),
```
Comment (alexkappa) [inline auth0/structure_auth0_connection.go:210]: Shouldn't this be flattened as well?

File: auth0/structure_auth0_connection.go [hunk 3]
```diff
@@ -485,6 +506,28 @@ func expandConnectionOptionsOIDC(d ResourceData) *management.ConnectionOptionsOI
...
 
+func expandConnectionOptionsSAML(d ResourceData) *management.ConnectionOptionsSAML {
+	o := &management.ConnectionOptionsSAML{
+		SigningCert:     String(d, "signing_cert"),
+		ProtocolBinding: String(d, "protocol_binding"),
+		IdpInitiated: &management.ConnectionOptionsSAMLIdpinitiated{
+			ClientID:             String(d, "client_id"),
+			ClientProtocol:       String(d, "client_protocol"),
+			ClientAuthorizequery: String(d, "client_authorizequery"),
+		},
```
Comment (alexkappa) [inline auth0/structure_auth0_connection.go:517]: I suspect there is an error here. `String(d, "client_id")` should be run inside a `Set(d, "idpinitiated").Elem(...)` block.

File: auth0/resource_auth0_connection_test.go [hunk 4]
```diff
@@ -1001,3 +1001,67 @@ func TestConnectionInstanceStateUpgradeV0(t *testing.T) {
...
+		protocol_binding = "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Post"
+		signature_algorithm = "rsa-sha256"
+		digest_algorithm = "sha256"
+		fields_map = {
+			foo = "bar"
+			baz = "baa"
+		}
+	}
+}
+`
```
Comment (alexkappa) [inline auth0/resource_auth0_connection_test.go:1067]: Could we add a test for updates? Some common issues tend to surface which get caught with update tests. Also please test the `idpinitiated` block.

File: auth0/resource_auth0_connection.go [hunk 5]
```diff
@@ -429,6 +429,78 @@ var connectionSchema = map[string]*schema.Schema{
...
+				"protocol_binding": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "The SAML Response Binding: how the SAML token is received by Auth0 from IdP",
+					ValidateFunc: validation.StringInSlice([]string{
+						"urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect",
+						"urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST",
+					}, true),
+				},
+				"idpinititated": {
```
Comment (alexkappa) [inline auth0/resource_auth0_connection.go:451]: We could make this a little more readable:
```suggestion
				"idp_inititated": {
```

File: auth0/resource_auth0_connection.go [hunk 6]
```diff
@@ -429,6 +429,78 @@ var connectionSchema = map[string]*schema.Schema{
...
+						Schema: map[string]*schema.Schema{
+							"client_id": {
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+							"client_protocol": {
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+							"client_authorizequery": {
```
Comment (alexkappa) [inline auth0/resource_auth0_connection.go:466]:
```suggestion
							"client_authorize_query": {
```

**Comment by alexkappa** [comment]
Awesome work @kgunbin thank you for this! Watch out for a new release soon.

**Comment by jbauth0** [comment]
Yes - thank you for all your work @kgunbin!

**Pull request merged**